src/server/compaction_filter_rule.h

```diff
@@ -36,5 +38,34 @@ class compaction_filter_rule
                        const std::string &sort_key,
                        const rocksdb::Slice &existing_value) const = 0;
 };
+
+enum string_match_type
+{
+    SMT_MATCH_ANYWHERE,
+    SMT_MATCH_PREFIX,
+    SMT_MATCH_POSTFIX,
+    SMT_INVALID,
+};
+ENUM_BEGIN(string_match_type, SMT_INVALID)
+ENUM_REG(SMT_MATCH_ANYWHERE)
+ENUM_REG(SMT_MATCH_PREFIX)
+ENUM_REG(SMT_MATCH_POSTFIX)
+ENUM_END(string_match_type)
+
+class hashkey_pattern_rule : public compaction_filter_rule
+{
+public:
+    explicit hashkey_pattern_rule(uint32_t pegasus_data_version);
```

Comment:
Why do hashkey pattern need `pegasus_data_version`? Is `pegasus_data_version` used for value schema?

Reply:
Yes, I can remove it in this pull request.